### ISSUE TITLE
vkCreateSwapchainKHR is not being called during window resizing under Wayland

### ISSUE THREAD

**Issue by charles-lunarg** [KhronosGroup/Vulkan-Samples#967]
While trying to capture various samples in gfxreconstruct, I noticed that vkCreateSwapchainKHR is not being called during resize events on Wayland on Ubuntu 23.10. This does not seem to be a functional problem for the samples, as resizing appears to work as expected. 

However, because vkCreateSwapchainKHR isn't being called, gfxreconstruct captures do not know that the swapchain is changing size, leading to the replay staying the same size. 

I create this issue in case best practice would be to recreate the swapchain when the window size changes, which is something that other presentation engines (eg xcb & win32) do require.

**Comment by charles-lunarg** [comment]
Just to clarify, while I do think it is a 'real' problem, this isn't a high priority issue. It was more frustrating to try to use Vulkan-Samples for some gfxrecon testing of wayland stuff, only to find out it didn't behave like I thought it would.

I don't think this affects too many users as it rendered correctly as far as I can tell.

**Comment by jeroenbakker-atmind** [comment]
I agree with @charles-lunarg as Wayland becomes more mature more users will eventually ask for this. It is quite some work to get everything similar to X11. We can give guidance to any volunteer who wants to dive into.

**Comment by jeroenbakker-atmind** [comment]
It was discussed to reach out to the vulkan community. Adding better support for wayland should have priority but is quit some work to get it correct. The idea is to reach out to devs that have done it already and check if they are willing to port their work over to the Vulkan samples repro.
